### ISSUE TITLE
qview (tracking) - add ability to copy and paste with cntl-c and cntl-v from one cell to another

### ISSUE THREAD

**Issue by ascbot** [DOI-USGS/ISIS3#1262]
---

Author Name: **Ella (Contractor) Lee** (Ella (Contractor) Lee)

Original Assignee: Kimberly Oyama

---

Add ability to select a cell, and copy and paste to other cells using ctrl-c and ctrl-v keys. I am currently having to type the values into each cell even though the contents is the same.  

*Additional information:*

transferred this post to category-applications from category-wishlist...seems like a reasonable feature request for qview that should be at a more exposed category level.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Kimberly Oyama** (Kimberly Oyama)
Original Date: 2012-06-20T16:15:42Z

---

I am going to update the help/documentation for the advanced tracking tool to include useful tool tips. The tips for the table will explain that the cell needs to be in the editing mode (double click on the cell, the cursor should be blinking) before the ctrl+c and ctrl+v commands work.

<B>IMPACT</B>
Anticipated impact to other parts of Isis is none.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Kris Becker** (Kris Becker)
Original Date: 2012-06-20T18:51:09Z

---

While your at it, the track tool table does not remember the column widths of cells from one invocation of qview to the next.  

Also there is a focus issue in that under certain circumstances, the track table disappears when you click on a different window other than the qview window.

Both of these issues occurs on the Mac. 

I can demonstrate them if needed.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Lynn Weller** (Lynn Weller)
Original Date: 2012-06-21T17:08:04Z

---

I can't get to this version of isis from astrovm1.  After setting isis to 
/work/projects/isis/latest/m00772/isis/ I try to launch qview and get this:

qview: error while loading shared libraries: libquadmath.so.0: cannot open shared object file: No such file or directory

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Kimberly Oyama** (Kimberly Oyama)
Original Date: 2012-06-22T19:31:33Z

---

Kris, since those issues are unrelated to this one, can you make another ticket to address them?

Lynn, you should be able to launch qview without any problems. You will get the updated version of the advanced tracking tool now.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Lynn Weller** (Lynn Weller)
Original Date: 2012-06-22T19:58:45Z

---

Ok, I can get to this version now and I see the tip/help.  There's a small typo under Table Help -- "youneed" needs a space.

One more thing, the help dialog comes up every time I open the tracking tool whether in the same session or is a start up a new session of qview.  Is it possible to see that only once, then have access to it somehow when necessary?
Thanks!

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Kimberly Oyama** (Kimberly Oyama)
Original Date: 2012-06-25T17:17:43Z

---

A help dialog box has been added. It appears the first time the tool is used and is available through the help menu (also added). The dialog has tips for recording and for using keyboard shortcuts in the table. If a tip is unclear please post a note with a more understandable alternative.

To Test:
Set Isis to: /work/projects/isis/latest/m00772/isis

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Lynn Weller** (Lynn Weller)
Original Date: 2012-06-25T17:59:38Z

---

Very nice - the tracking tool help is much more user friendly now.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Ella (Contractor) Lee** (Ella (Contractor) Lee)
Original Date: 2012-06-25T22:36:08Z

---

I like the help button.  Suggestion to make it clear you also need to double click on the cell you are pasting the copied contents to.

In order to use ctrl+c to copy and ctrl+v to paste, you need to double click on the cell (only the text should be highlighted) you are copying from and then double click on the cell (should see a cursor) you are pasting to.  If the entire cell is highlighted then it is not in editing mode.  When a cell is in this editing mode, most keyboard shortcuts work.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Kimberly Oyama** (Kimberly Oyama)
Original Date: 2012-07-03T22:37:52Z

---

I've updated the help text to reflect your suggestion. Please let me know if there are any other changes you'd like me to make.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Ella (Contractor) Lee** (Ella (Contractor) Lee)
Original Date: 2012-07-05T21:55:24Z

---

The help information added to the tracking section is clearer and useful.

Thanks,
Ella Mae
